src/analysis/bounds.rs

```diff
@@ -154,12 +160,49 @@ impl Bounds {
                             bound_name,
                         });
                     }
+                } else if par.c_type.ends_with("Func") || par.c_type.ends_with("Callback") ||
+                          par.c_type == "GDestroyNotify" {
+                    need_is_into_check = par.c_type != "GDestroyNotify";
+                    if let Type::Function(_) = env.library.type_(par.typ) {
+                        type_string = rust_type_with_scope(env, par.typ, par.scope).into_string();
+                        let bound_name = *self.unused.front().unwrap();
+                        callback_info = Some(CallbackInfo {
+                            callback_type: type_string.clone(),
+                            success_parameters: String::new(),
+                            error_parameters: String::new(),
+                            bound_name,
+                        });
+                    }
                 }
-                if !self.add_parameter(&par.name, &type_string, bound_type, async) {
-                    panic!(
-                        "Too many type constraints for {}",
-                        func.c_identifier.as_ref().unwrap()
-                    )
+                if par.c_type != "GDestroyNotify" {
+                    if !self.add_parameter(&par.name, &type_string, bound_type, async) {
+                        panic!(
+                            "Too many type constraints for {}",
+                            func.c_identifier.as_ref().unwrap()
+                        )
+                    }
+                }
+                if need_is_into_check {
+                    if let Some(x) = if let Some(ref mut last) = self.used.last_mut() {
+                        if last.bound_type.is_into() {
+                            let mut new_one = (*last).clone();
+                            new_one.alias = self.unused.pop_front().expect("no available bound");
+                            new_one.type_str = last.alias.to_string();
+                            new_one.parameter_name = last.parameter_name.clone();
+                            new_one.hack = true;
```

Comment:
This is the `hack`, so I guess a comment above here would also be useful to explain for the future what exactly the code is trying to do and why